client/lib/signup/step-actions.js

```diff
@@ -697,6 +698,26 @@ export function isSiteTopicFulfilled( stepName, defaultDependencies, nextProps )
 	}
 }
 
+export function isSiteStyleFulfilled( stepName, defaultDependencies, nextProps ) {
+	// Switching to a site type where the mockups are hidden means we
+	// don't want to ask what site style they want, we'll pick one for them.
+	if ( getSiteTypePropertyValue( 'slug', nextProps.siteType, 'hideSiteMockups' ) ) {
+		const { siteType } = nextProps;
+		const { id: siteStyle, theme: themeSlugWithRepo } = getSiteStyleOptions( siteType )[ 0 ];
+
+		nextProps.submitSignupStep( { stepName }, { siteStyle, themeSlugWithRepo } );
+		flows.excludeStep( stepName );
+	}
+}
+
+export function isSiteStyleUnfulfilled( stepName, defaultDependencies, nextProps ) {
```

Comment:
I have a few thoughts around these two functions:

Firstly, if I were to see any function with the format `isXX` without looking in to the functions implementation, I'd make some assumptions:
- It returns a boolean
- It should be pure - it invokes no side effects, it is 100% deterministic based on state or inputs

So I think these two should be renamed to something that better expresses their usage...

I see that other, similar functions in `step-actions` - `isDomainFulfilled`, `isPlanFulfilled` etc. - follow this same naming pattern. Those are improperly named in my opinion ([Also, see our guidelines here](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#function-names)).

Both are used like so (this is sort of just a note to self / showing my thoughts):
```js
'site-style': {
  stepName: 'site-style',
  ...,
  fulfilledStepCallback: isSiteStyleFulfilled,
  unfulfilledStepCallback: isSiteStyleUnfulfilled,
},
```

There's a confusing logic train to follow in order to properly understand where these callbacks get called... Here's my understanding of that, with notes:
```
➥ `signup/main.jsx` - mount & update lifecycle methods
  ➥ removeFulfilledSteps
      - Note: This also feels wrongly named - It doesn't actually remove anything?! 
        Instead it invokes callbacks and goes to a particular step under conditions.
        Those callbacks may do some 'removing', but how can we make that assumption here?
    ➥ processFulfilledSteps
        - Note: Only Handles a single step, should be called processFulfilledStep 
        - Note: 'process' here is too generic, not particularly descriptive of it's responsibility
        - Note: Really, there's very little point in this being it's own method - more indirection with zero upside, could instead just be defined as the predicate of the loop in `removeFulfilledSteps`
      ➥ Invokes `fulfilledStepCallback` on the step if it is not excluded and has a `fulfilledStepCallback` prop
```

We're getting in to this flow every time the component updates too... I've not really explored this mentally yet but my gut tells me that we shouldn't be doing this so liberally, especially as we're invoking callbacks in such a non-obvious way.

I'm not sure what my suggestion here is, other than I'd really love to see us reduce this complexity first before we add to it.
is there a way to achieve the same goals, without adding more indirection?

Reply:
Agree with everything here